x/appchain/coordinator/keeper/ibc_client.go

```diff
@@ -0,0 +1,239 @@
+package keeper
+
+import (
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	"github.com/ExocoreNetwork/exocore/utils"
+	commontypes "github.com/ExocoreNetwork/exocore/x/appchain/common/types"
+	"github.com/ExocoreNetwork/exocore/x/appchain/coordinator/types"
+	avstypes "github.com/ExocoreNetwork/exocore/x/avs/types"
+	abci "github.com/cometbft/cometbft/abci/types"
+	tmtypes "github.com/cometbft/cometbft/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	commitmenttypes "github.com/cosmos/ibc-go/v7/modules/core/23-commitment/types"
+	ibctmtypes "github.com/cosmos/ibc-go/v7/modules/light-clients/07-tendermint"
+)
+
+// CreateClientForSubscriberInCachedCtx is a wrapper function around CreateClientForSubscriber.
+func (k Keeper) CreateClientForSubscriberInCachedCtx(
+	ctx sdk.Context,
+	req types.RegisterSubscriberChainRequest,
+) (cctx sdk.Context, writeCache func(), err error) {
+	cctx, writeCache = ctx.CacheContext()
+	err = k.CreateClientForSubscriber(cctx, req)
+	return
```

Comment:
should the `writeCache` be called before return if the err is nil?

Reply:
`writeCache` and `err` processing is handled by the caller. This is why the function has the prefix `InCachedCtx`.